horovod/spark/common/util.py

```diff
@@ -695,3 +695,13 @@ def to_list(var, length):
                              "label_cols")
 
     return var
+
+
+def _get_assigned_gpu_or_default(default):
+    try:
+        # if GPU-aware scheduling is available, pin to the assigned GPU index
+        from horovod.spark.task import get_available_devices
+        return int(get_available_devices()[0])
```

Comment:
What exception do you expect this call to raise? If it returns an empty list and `[0]` throws an index out of bound error, then it is better to do:

~~~python
from horovod.spark.task import get_available_devices
available_devices = get_available_devices()
if available_devices:
  return int(available_devices[0])
else:
  return default
~~~

The difference is that if the cluster has GPU-aware configured but GPU index was done wrongly, e.g., "gpu1" instead of "1". We should throw an exception instead of fallback.